### PR TITLE
Add support for selected overdue appointments download

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/8665/add-support-for-selected-overdue-appointments-download

### CODE REVIEW

**Comment by janhavisinghh** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4067** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4067" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/4067?utm_source=stack-comment).

File: app/src/androidTest/java/org/simple/clinic/overdue/download/OverdueListDownloaderIntegrationTest.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ class OverdueListDownloaderIntegrationTest {
   fun downloading_a_csv_should_work_correctly() {
     // when
     val result = overdueListDownloader
-        .download(OverdueListFileFormat.CSV)
+        .download(OverdueListFileFormat.CSV, listOf(UUID.fromString("762f1a02-76aa-4dd7-8669-48aa0deac64b")))
```
Comment (msasikanth) [inline app/src/androidTest/java/org/simple/clinic/overdue/download/OverdueListDownloaderIntegrationTest.kt:37]: I didn't add the selected ids here as there are couple of new tests that handle ids. Here I was verifying it loads from the server API. So, pass an empty list here?

File: app/src/androidTest/java/org/simple/clinic/overdue/download/OverdueListDownloaderIntegrationTest.kt [hunk 2]
```diff
@@ -45,7 +45,7 @@ class OverdueListDownloaderIntegrationTest {
   fun downloading_a_pdf_should_work_correctly() {
     // when
     val result = overdueListDownloader
-        .download(OverdueListFileFormat.PDF)
+        .download(OverdueListFileFormat.PDF, listOf(UUID.fromString("762f1a02-76aa-4dd7-8669-48aa0deac64b")))
```
Comment (msasikanth) [inline app/src/androidTest/java/org/simple/clinic/overdue/download/OverdueListDownloaderIntegrationTest.kt:48]: Same comment as above.

File: app/src/main/java/org/simple/clinic/overdue/download/OverdueDownloadWorker.kt [hunk 3]
```diff
@@ -80,9 +86,10 @@ class OverdueDownloadWorker(
 
     val downloadFormatString = inputData.getString(KEY_DOWNLOAD_FORMAT)!!
     val downloadFormat = OverdueListFileFormat.valueOf(downloadFormatString)
+    val selectedAppointmentIds = inputData.getStringArray(KEY_SELECTED_APPOINTMENT_IDS)?.map { UUID.fromString(it) }.orEmpty()
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/overdue/download/OverdueDownloadWorker.kt:89]: FYI, we can use `map(UUID::fromString)` to make it a bit concise.

**Review by msasikanth — CHANGES_REQUESTED** [review]
Overall it looks good, are we going to pass these IDs from the screens in a different PR?

**Review by msasikanth — APPROVED** [review]

**Pull request merged**